### PR TITLE
Invariants feat: add config option to turn off shrinking

### PR DESCRIPTION
**Pull request by grandizzy**
## Motivation

See https://github.com/foundry-rs/foundry/issues/4867, when running invariants test with many runs and depth (1000) the test hangs in shrinking scenario (performed for each invariant). In this case sometimes is preferable to fail fast and address the issue.

## Solution

* add `invariant_shrink_sequence` option, default true, tries to reduce number of calls in scenario to min
* when set to false test will fail fast and output the original number of calls in scenario

### CODE REVIEW

**Comment by mds1** [comment]
@0xmelkor note that we'll need to update https://github.com/foundry-rs/foundry/pull/4744 to support this option in there too

**Comment by 0xMelkor** [comment]
> @0xMelkor note that once this is merged we'll need to update #4744 to support this option in there too

Hi @mds1, sure.

File: evm/src/fuzz/invariant/error.rs [hunk 1]
```diff
@@ -90,21 +94,25 @@ impl InvariantFuzzError {
...
             TestError::Abort(_) => return Ok(None),
-            TestError::Fail(_, ref calls) => calls,
+            TestError::Fail(_, ref _calls) => _calls,
         };
 
-        let calls = self.try_shrinking(calls, &executor);
+        if self.shrink {
+            let _calls = self.try_shrinking(_calls, &executor);
+        } else {
+            trace!(target: "forge::test", "Shrinking disabled.");
```
Comment (mattsse) [inline evm/src/fuzz/invariant/error.rs:106]: please remove `_` prefix

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit, otherwise lgtm

**Comment by grandizzy** [comment]
> smol nit, otherwise lgtm

yep, added such to avoid `warning: unused variable: if this is intentional, prefix it with an underscore` is there a different preferred way to disable this?

**Comment by mattsse** [comment]
but 

> `let _calls =`

`let _calls = self.try_shrinking(_calls, &executor);`

so I think it should be `let calls = ...; let _ self.try_shrinking(calls,&executor)`

**Comment by grandizzy** [comment]
> but
> 
> > `let _calls =`
> 
> `let _calls = self.try_shrinking(_calls, &executor);`
> 
> so I think it should be `let calls = ...; let _ self.try_shrinking(calls,&executor)`

oops, right, changing

**Comment by grandizzy** [comment]
> > but
> > > `let _calls =`
> > 
> > 
> > `let _calls = self.try_shrinking(_calls, &executor);`
> > so I think it should be `let calls = ...; let _ self.try_shrinking(calls,&executor)`
> 
> oops, right, changing

fixed with https://github.com/foundry-rs/foundry/pull/4868/commits/022646173af48b63cbf76d98ebf1a90d1039c597

**Review by mattsse — APPROVED** [review]
cool, tysm!

lgtm!

File: config/src/invariant.rs [hunk 2]
```diff
@@ -61,6 +64,7 @@ impl InlineConfigParser for InvariantConfig {
                 "depth" => conf_clone.depth = parse_config_u32(key, value)?,
                 "fail-on-revert" => conf_clone.fail_on_revert = parse_config_bool(key, value)?,
                 "call-override" => conf_clone.call_override = parse_config_bool(key, value)?,
+                "shrink-sequence" => conf_clone.shrink_sequence = parse_config_bool(key, value)?,
```
Comment (Evalir) [inline config/src/invariant.rs:67]: added the new `shrink-sequence` key to the invariant inlineconfigparser so it works with #4744

**Review by Evalir — APPROVED** [review]
LGTM! Just pushed something upstream to make it compatible with #4744. Nice work!

**Review by mattsse — APPROVED** [review]

**Comment by grandizzy** [comment]
thank you all!

**Pull request merged**